ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml

```diff
@@ -110,31 +110,50 @@ let checksum rpc session_id vdi =
       Digest.to_hex (Digest.file path)
     )
 
-let check_vdi_unchanged session_id ~prepare_vdi sr_info =
+let check_vdi_unchanged session_id ~prepare_vdi ~vdi_op sr_info =
   let rpc = !Quicktest_common.rpc in
   let sR = sr_info.Storage_test.sr in
-  let vdi = Client.Client.VDI.create ~rpc ~session_id ~name_label:"" ~name_description:"" ~sR ~virtual_size:4194304L ~_type:`user ~sharable:false ~read_only:false ~other_config:[] ~xenstore_data:[] ~sm_config:[] ~tags:[] in
-  Storage_test.VDI.with_destroyed session_id vdi (fun () ->
+  Storage_test.VDI.with_new session_id sR (fun vdi ->
       prepare_vdi vdi;
       let checksum_original = checksum rpc session_id vdi in
-      let copy = Client.Client.VDI.copy ~rpc ~session_id
-          ~vdi
-          ~base_vdi:API.Ref.null
-          ~into_vdi:API.Ref.null
-          ~sr:sR
-      in
-      Storage_test.VDI.with_destroyed session_id copy (fun () ->
-          let checksum_copy = checksum rpc session_id vdi in
+      let new_vdi = vdi_op rpc session_id sR vdi in
+      Storage_test.VDI.with_destroyed session_id new_vdi (fun () ->
+          let checksum_copy = checksum rpc session_id new_vdi in
```

Comment:
Was this buggy previously? Was it checksumming the same VDI and comparing it to itself? Nice spot.

Reply:
Yep, it was, it was checking that `a = a` :flushed: